packages/components/src/components/buttonlink/buttonlink.component.ts

```diff
@@ -0,0 +1,177 @@
+import { CSSResult, html, PropertyValues } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import styles from './buttonlink.styles';
+import Link from '../link';
+import {
+  BUTTON_COLORS,
+  BUTTON_TYPE_INTERNAL,
+  BUTTON_VARIANTS,
+  DEFAULTS,
+  ICON_BUTTON_SIZES,
+  PILL_BUTTON_SIZES,
+} from '../button/button.constants';
+import type { IconNames } from '../icon/icon.types';
+import type { ButtonColor, ButtonVariant, IconButtonSize, PillButtonSize } from '../button/button.types';
+import type { LinkSize } from '../link/link.types';
+import Button from '../button/button.component';
+import type { ButtonLinkTypeInternal } from './buttonlink.types';
+
+/**
+ * `mdc-buttonlink` is a link that looks like a button.
+ * It supports all features of `mdc-link`, along with `prefix-icon` and `postfix-icon` slots.
+ *
+ * @dependency mdc-icon
+ *
+ * @tagname mdc-buttonlink
+ *
+ * @event click - (React: onClick) Fired when the user activates the buttonLink using a mouse or assistive technology.
+ * @event keydown - (React: onKeyDown) Fired when the user presses a key while the buttonLink has focus.
+ * @event focus - (React: onFocus) Fired when the buttonLink receives keyboard or mouse focus.
+ * @event blur - (React: onBlur) Fired when the buttonLink loses keyboard or mouse focus.
+ *
+ */
+class Buttonlink extends Link {
```

Comment:
```suggestion
class ButtonLink extends Link {
```

Please make this change across all instances. We have this as a part of our [conventions](https://github.com/momentum-design/momentum-design/blob/main/packages/components/conventions/component-naming-guidelines.md).

Reply:
Agreed but, I intentionally went with the name Buttonlink for these reasons:
✅ Buttonlink → “A link that looks like a button” – correct intent, simple and smooth naming.
🚫 ButtonLink → “A half-button, half-link” – misleading, as it overemphasizes the ‘button’ aspect, whereas the core behavior is still that of a link.

Reply:
Since we have this as a convention, we'd need to follow it for maintaining consistency.
I differ to agree that by changing the case, it would provide a different meaning.
We are doing the same on `AvatarButton` too. This component is intended to be a button and look like an avatar.

Reply:
Initially, I felt AvatarButton should also be named as Avatarbutton as per my reasoning. 
Okay I understand, it's not very much important for consumers though so let's stick to what we have defined so far.